### DISCUSSION TITLE
Making camera jump to player instantly.

### DISCUSSION THREAD

**Discussion by tetraminus** [ramokz/phantom-camera#229] [Question / Help Wanted]
When my game starts, the player is randomly placed into the level and the camera is set to follow it. The problem is, that the camera has to tween to the player from its starting point, this technically happens post load, so the tween on load option doesn't seem to affect it. is there a way to skip the tween at any given moment? I'm using CS but the general idea should be shared across languages.

This is my current code. my thought was to set the tween time to 0, wait a frame, the set it back, but it doesn't work, any ideas? 
```cs
public void AddPlayer(Player.Player player)
    {
        AddChild(player);
        player.GlobalPosition = GetSpawnPosition(true);
        
        var pcamProperties = PhantomCamera.Get("Properties").AsGodotObject();
        
        PhantomCamera.Set("follow_target", player.GetPath());
        
        
        pcamProperties.Set("has_tweened", true);
        PhantomCamera.Get("tween_parameters").AsGodotObject().Set("duration", 0);
        PhantomCamera.GlobalPosition = player.GlobalPosition;
        FixCameraTween();
       
        
        player.Show();
        
    }
    
    public async void FixCameraTween()
    {
       await ToSignal(GetTree(), SceneTree.SignalName.ProcessFrame);
       PhantomCamera.Get("tween_parameters").AsGodotObject().Set("duration", 1);
    }```

**Comment by ramokz** [comment]
If you have the `tween on load` disabled, then the tweening should only happen if you're moving from one `PCam` to another as the priority value changes. Even if you have another `PCam` in the scene before the player's `PCam` gets instantiated, it should still jump straight to the Player `PCam`.

https://github.com/ramokz/phantom-camera/assets/5159399/2a79eec1-5ab0-43d4-bf5f-6b69d170b1bb

`PlayerPCam` with `tween on load` **on**.

---

https://github.com/ramokz/phantom-camera/assets/5159399/725a5839-75eb-4060-8fff-7e552b93b6e0

`PlayerPCam` with `tween on load` **off**.

---

Both examples are using a default tween resource, but changing it didn't affect the result.

My only guess would be that it's the `follow damping` that's causing the issue.

**Comment by tetraminus** [comment]
I managed to fix it by doing the same wait for 2 frames and switching the damping off and on.
```cs
public async void FixCameraTween()
    {
        // Going to hell for this
       await ToSignal(GetTree(), SceneTree.SignalName.ProcessFrame);
       await ToSignal(GetTree(), SceneTree.SignalName.ProcessFrame);
       PhantomCamera.Get("tween_parameters").AsGodotObject().Set("duration", 1);
       PhantomCamera.Set("follow_parameters/damping", true);
   }
```
probably not the best way to do it but it works for now?

**Reply by ramokz** [reply]
This _shouldn't_ be necessary.

Don't see you're changing the priority value in your code sample, so am guessing you only have one `PCam` in the scene - that being the `PlayerPCam`?

If so, is there a reason why you're not just adding the `PlayerPCam` to the Player scene that you're instantiating?

**Reply by tetraminus** [reply]
having the PlayerPCam be part of the player makes it follow the player rigidly, ignoring damping. I assume because it's always following the parent?

**Reply by ramokz** [reply]
Correct, but you can add a parent node, or a scene root, to the scene you're instantiating.
So it would look something like this:

- PlayerSceneRoot (Node2D/Node3D)
  - Player (the one you're currently instantiating)
  - PlayerPCam

That way the `PlayerPCam` won't be tied to the Player's movement.

**Reply by tetraminus** [reply]
That makes sense, I'm going try that setup.
